File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/core/PersistedTypes.scala
package dev.typr.foundationssc.docs.core
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*

@SuppressWarnings(Array("unused"))
object PersistedTypes:
  //start
  case class VenueId(value: Long)
  val venueIdType: DuckDbType[VenueId] = DuckDbTypes.bigint.transform(VenueId.apply, _.value)

  case class Venue(name: String, capacity: Int)
  case class PersistedVenue(id: VenueId, venue: Venue)
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/PersistedTypes.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.*;

@SuppressWarnings("unused")
public class PersistedTypes {
    //start
    public record VenueId(Long value) {}
    static DuckDbType<VenueId> venueIdType = DuckDbTypes.bigint.transform(VenueId::new, VenueId::value);

    public record Venue(String name, int capacity) {}
    public record PersistedVenue(VenueId id, Venue venue) {}
    //stop
}
